lib/elixir/src/elixir_erl_compiler.erl

```diff
@@ -74,10 +74,6 @@ handle_file_error(File, {Line, Module, Desc}) ->
 
 %% Custom formatting
 
-%% Normalize formatting of functions
-format_error(erl_lint, {undefined_function, {F, A}}) ->
-  io_lib:format("undefined function ~ts/~B", [F, A]);
```

Comment:
I think the error is still coming from erl_lint, no? It just happens that now we are using the default formatting instead of our custom one.

Reply:
Sorry, I got confused hahaha

I'll work on it

Reply:
I was also investigating this issue a few weeks ago before going on vacation, and I was planning on working on it today 😅 But don't worry, I didn't go so far. Last thing I remember was reading and understanding the code of `Kernel.LexicalTracker` (which might not even be related to the issue! 😬).

If you don't mind, let's "call dibs" on the issues of #5800 so we don't duplicate work 🙂

Reply:
I don't think it is the Kernel.LexicalTracker. Most likely the Module.LocalsTracker or something similar. :)

Reply:
Sorry, it was `Module.LocalsTracker`. [There is a "lightweight" digraph using ETS](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/module/locals_tracker.ex#L181-L205), and I thought the solution might be there...or not 😅

Reply:
@josevalim you're always faster than me 🙂 (I remembered the name was something with "tracker" 😬 )

Reply:
Yeah, I was playing with the `LocalsTracker` but when I saw the tests were failing only on the string I thought the error where dealt somewhere else.  I've pushed a working version now